pkg/querier/querier.go

```diff
@@ -132,9 +133,9 @@ func ShouldQueryBlockStore(queryStoreAfter time.Duration, now time.Time, queryMi
 func New(cfg Config, limits *validation.Overrides, distributor Distributor, storeQueryable storage.Queryable, reg prometheus.Registerer, logger log.Logger, tracker *activitytracker.ActivityTracker) (storage.SampleAndChunkQueryable, storage.ExemplarQueryable, *promql.Engine) {
 	queryMetrics := stats.NewQueryMetrics(reg)
 
-	distributorQueryable := newDistributorQueryable(distributor, mergeChunks, limits, queryMetrics, logger)
+	distributorQueryable := newDistributorQueryable(distributor, batch.NewChunkMergeIterator, limits, queryMetrics, logger)
```

Comment:
Why passing the function at all to `newDistributorQueryable()`?

Reply:
good call. I can hardocde it further down.